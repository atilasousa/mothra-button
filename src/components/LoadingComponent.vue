<template>
    <div class="loader" :class="getSize()"></div>
</template>
<script setup lang="ts">

const props = withDefaults(defineProps<{
    size?: "md" | "sm" | "lg";
}>(), {
    size: "md",

})

function getSize(): string {
    return `loader-${props.size}`
}
</script>
<style scoped lang="scss">
$background: #ffffff50;
$circle: #ffffff;

.loader {
    border: 4px solid $background;
    border-radius: 50%;
    border-top: 4px solid $circle;
    width: 12px;
    height: 12px;
    -webkit-animation: spin 2s linear infinite;
    animation: spin 1s linear infinite;
}

.loader-sm {
    border: 3px solid $background;
    border-top: 3px solid $circle;
    width: 10px;
    height: 10px;
}

.loader-lg {
    border: 5px solid $background;
    border-top: 5px solid $circle;
    width: 15px;
    height: 15px;
}

@-webkit-keyframes spin {
    0% {
        -webkit-transform: rotate(0deg);
    }

    100% {
        -webkit-transform: rotate(360deg);
    }
}

@keyframes spin {
    0% {
        transform: rotate(0deg);
    }

    100% {
        transform: rotate(360deg);
    }
}
</style>