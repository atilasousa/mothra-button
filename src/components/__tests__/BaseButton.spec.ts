import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import BaseButton from '../BaseButton.vue'
import LoadingComponent from '../LoadingComponent.vue'



describe('BaseButton', () => {
  it('Checking if component is rendered', () => {
      const wrapper = mount(BaseButton)
      expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('BaseButton / Checking if slots are being render',()=>{
  it('Check if icon slot is being rendered', () => {
    const wrapper = mount(BaseButton, { slots: { icon: 'icon' } })
    expect(wrapper.html()).toContain('icon')
  })
  it('Check if label slot is being rendered', () => {
    const wrapper = mount(BaseButton, { slots: { label: 'Button' } })
    expect(wrapper.html()).toContain('Button')
  })
})
 
describe('BaseButton / Checking Component Size', () => {
  it('Check if button medium size is render', () => {
    const wrapper = mount(BaseButton, { props: { size: 'md' } })
    expect(wrapper.classes()).toContain('cf-button-md')
  })
  it('Check if button Small size is render', () => {
    const wrapper = mount(BaseButton, { props: { size: 'sm' } })
    expect(wrapper.classes()).toContain('cf-button-sm')
  })
  it('Check if button Large size is render', () => {
    const wrapper = mount(BaseButton, { props: { size: 'lg' } })
    expect(wrapper.classes()).toContain('cf-button-lg')
  })
})

describe('BaseButton / Checking Button Status',()=>{
  it('Check if button is disabled', () => {
    const wrapper = mount(BaseButton, { props: { isDisabled: true } })
    expect(wrapper.attributes("disabled")).toBe("")
  })
  it('Check if button is enabled', () => {
    const wrapper = mount(BaseButton, { props: { isDisabled: false } })
    expect(wrapper.attributes("disabled")).toBeUndefined()
  })
})

describe('BaseButton / Checking Icon Position',()=>{
  it('Check if the icon is positioned in the right position (The icon will only be positioned on the right, if exist a label and a icon slot")', () => {
    const wrapper = mount(BaseButton, {slots: { label: 'Button',icon:"icon" }, props: { iconPosition: 'right' } });
    expect(wrapper.classes()).toContain("cf-button-icon-right")
  })
  it('Check if button is positioned in the left position (The icon will only be positioned on the left, if exist a label and a icon slot")', () => {
    const wrapper = mount(BaseButton, {slots: { label: 'Button',icon:"icon" }, props: { iconPosition: 'left' } });
    expect(wrapper.classes()).toContain("cf-button-icon-left")
  })
  it('Check if the icon is the only content present in the button")', () => {
    const wrapper = mount(BaseButton, {slots: { label: 'Button',icon:"icon" }, props: { iconPosition: 'onlyIcon' } });
    expect(wrapper.html()).toContain('icon')
    expect(wrapper.find('.cf-button-label').exists()).toBe(false)
  })
})

describe('BaseButton / Checking if component is loading',()=>{
  it('check if LoadingComponent is rendered',()=>{
    const wrapper = mount(BaseButton,{props:{loading:true}});
    const loadingComponent=wrapper.findComponent(LoadingComponent);
    expect(loadingComponent.exists()).toBe(true);
  })
})

