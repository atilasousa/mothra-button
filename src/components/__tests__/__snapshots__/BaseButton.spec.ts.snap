// Vitest Snapshot v1

exports[`BaseButton > Checking if component is rendered 1`] = `
"<button class=\\"cf-button cf-button-md\\">
  <div class=\\"cf-button-icon\\"></div><span class=\\"cf-button-label\\"></span>
  <!--v-if-->
</button>"
`;
